### ISSUE TITLE
Some backgrounds shifted over to the right by 1, wrap onto left side

### ISSUE THREAD

**Issue by MP2E** [MiSTer-devel/GBA_MiSTer#5]
[The 240p Test Suite for GBA](https://github.com/pinobatch/240p-test-mini/releases) has an overscan test that reveals some minor problems. The bottom row of pixels is cut off, and the picture is shifted over to the right by 1 pixel, with the far most right side wrapping over to the left side of the screen.

The strange thing is that not all GBA games draw the right most line on the left most line, in fact, most seem to not do this.

Here's a screenshot:
![Screenshot_from_2019-11-24_19-04-10](https://user-images.githubusercontent.com/167708/69516752-73c99580-0f07-11ea-8912-f27a91232270.png)

Since most GBA games don't draw the rightmost line on the left, I looked into what this test is doing, and it seems like it is using BG1 for the outside, so maybe the issue lies there.

Reference: https://github.com/pinobatch/240p-test-mini/blob/master/gba/src/overscan.c#L127

**Comment by sorgelig** [comment]
i will check it.

**Comment by sorgelig** [comment]
Agree that bottom row is hidden - i will check it.
But there is no wrapping. Scroll tests in Test Suite don't show any wrapping. Also games drawing from edge to edge like Mario Kart look correct without wrappings.

**Comment by MP2E** [comment]
Yeah the wrapping seems to only happen with the overscan test, I thought the test might be wrong but it doesn't do this in mGBA. Not sure why there would be a difference.

But in games and the scrolling test there is no wrapping, like you said.

**Comment by ghogan42** [comment]
We tried to figure out the bottom row on discord today and could not. I found this:

(1) video_info popup thinks we are doing 240x160
(2) Blanking code looked like it was doing 160 lines but is hard to follow.

But when you enable vscale_mode=1 to get 6x integer scaling

Some rows are 6 pixels high and some rows are 7 pixels high. At the bottom of the screen this has happened several times and there is not enough space to have the 160th row.

Maybe there are 160 rows but something is making the scaler work incorrectly?

The Test Suite Left hand column bug looks like it's limited to that one test. It happens on MiSTer but not in an emulator.  Maybe some unrelated bug in the core.

**Comment by paulb-nl** [comment]
Noticed the shift in the Advance wars intro
![advancewars](https://user-images.githubusercontent.com/2227984/69534228-a1e7bd80-0f79-11ea-81e9-4df243e930fb.jpg)

**Comment by sorgelig** [comment]
These sporadic shifts seems related to internal rendering, not to frame buffer.

**Comment by sorgelig** [comment]
what i notice on grid test that bottom row of squares are not really square but stretched vertically. Could it be a test problem?

**Comment by MP2E** [comment]
![Screenshot from 2019-11-25 12-59-23](https://user-images.githubusercontent.com/167708/69577561-6f42c280-0f83-11ea-8aa3-a0fa5d4894dc.png)

Here's what the grid test looks like on my Game Boy Player and flash cart. I think it's a core or scaler issue

**Comment by RobertPeip** [comment]
fixed
